app/controllers/galleries_controller.rb

```diff
@@ -101,12 +99,10 @@ def edit
   def update
     @gallery.assign_attributes(gallery_params)
 
-    begin
-      Gallery.transaction do
-        @gallery.gallery_groups = process_tags(GalleryGroup, :gallery, :gallery_group_ids)
-        @gallery.save!
-      end
-    rescue ActiveRecord::RecordInvalid
+    if @gallery.update(gallery_groups: process_tags(GalleryGroup, :gallery, :gallery_group_ids))
```

Comment:
This is harder to read like this, imo, since `.update(...)` looks like it's going to update _only_ the things you put there (not things you might have also changed earlier). I think it'll work, but I'd prefer to either do the explicit transaction here or else something like:

```rb
updates = gallery_params.merge(gallery_groups: process_tags(GalleryGroup, :gallery, :gallery_group_ids))
if @gallery.update(updates)
```